src/main/java/org/dmfs/iterators/SerialIterator.java

```diff
@@ -43,7 +43,7 @@
     @SafeVarargs
     public SerialIterator(final Iterator<E>... iterators)
     {
-        mIterators = iterators.clone();
+        mIterators = iterators;
```

Comment:
Ok, I've simply removed that optimization now. (I also see that the class is deprecated.)